pkg/registry/core/rest/storage_core.go

```diff
@@ -201,7 +201,7 @@ func (c LegacyRESTStorageProvider) NewLegacyRESTStorage(restOptionsGetter generi
 	serviceClusterIPAllocator, err := ipallocator.New(&serviceClusterIPRange, func(max int, rangeSpec string) (allocator.Interface, error) {
 		mem := allocator.NewAllocationMap(max, rangeSpec)
 		// TODO etcdallocator package to return a storage interface via the storageFactory
-		etcd, err := serviceallocator.NewEtcd(mem, "/ranges/serviceips", api.Resource("serviceipallocations"), serviceStorageConfig)
+		etcd, err := serviceallocator.NewEtcd(mem, "/ranges/serviceips", api.Resource("serviceipallocations"), &serviceStorageConfig.Config)
```

Comment:
I would prefer changing the NewEtcd function to instead take ConfigForResource.

Reply:
What about the GroupResource parameter?  Does `serviceallocator.NewEtcd` do anything with it besides invoke its `String()`?

Reply:
It blindly passes this down: https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/service/allocator/storage/storage.go#L79

Reply:
In the revision you reviewed, https://github.com/MikeSpreitzer/kubernetes/blob/963959f8e31be9871466e530a34d67f67a0404db/pkg/registry/core/service/allocator/storage/storage.go#L65 also used the GroupResource:
```
	storage, d, err := generic.NewRawStorage(config.ForResource(resource), nil)
```

I have revised as I think you were suggesting and we discussed yesterday.